### PR TITLE
Fix loading data to Tensor from array-like objects

### PR DESCRIPTION
**Pull request by mmikolajcz**
### Details:
 - Fix loading data to Tensor from array-like objects (tf.Tensor, torch Tensor) in Python API
 - Add tests cases for array-like objects

### Tickets:
 - 86598

### CODE REVIEW

File: src/bindings/python/src/openvino/runtime/ie_api.py [hunk 1]
```diff
@@ -179,13 +186,14 @@ def start_async(
 
         (1) `numpy.array`
         (2) `openvino.runtime.Tensor`
+        (3) array-like object with __array__ attribute
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/ie_api.py:189]:
```suggestion
        (3) array-like object with `__array__` attribute
```

File: src/bindings/python/src/openvino/runtime/ie_api.py [hunk 2]
```diff
@@ -297,13 +308,14 @@ def start_async(
...
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/ie_api.py:311]:
```suggestion
        (3) array-like object with `__array__` attribute
```

**Review by jiwaszki — APPROVED** [review]
Just fix style warnings, overall looks good! Great job 🥇

File: src/bindings/python/src/openvino/runtime/ie_api.py [hunk 3]
```diff
@@ -125,13 +128,14 @@ def infer(self, inputs: Union[dict, list, tuple, Tensor, np.ndarray] = None) ->
...
```
[the remark this answers is not in the document]
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/ie_api.py]: Fixed

File: src/bindings/python/src/openvino/runtime/ie_api.py (hunk 1 above)
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/ie_api.py]: Fixed

File: src/bindings/python/src/openvino/runtime/ie_api.py (hunk 2 above)
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/ie_api.py]: Fixed

**Review by helena-intel — APPROVED** [review]
Thank you! Very happy to see this. I tested this and my code that uses PyTorch tensors as inputs works again.

**Review by p-wysocki — APPROVED** [review]
Looks good, I really like the idea of checking `__array__` attribute.

**Pull request merged**